src/merge_sort.py

```diff
@@ -0,0 +1,64 @@
+"""A merge sort function."""
+from insertion_sort import existing_set_insertion_sort as esis
+
+from insertion_sort import insertion_sort
+
+
+def merge_sort(numbers):
+    """Sort a iterable of numbers into min first via merge sort."""
+    final = []
+    compare = []
+    temp = []
+    hold_extra = []
+    if len(numbers) == 1:
+        return numbers
+    elif len(numbers) == 2:
+        if numbers[0] > numbers[1]:
+            numbers[0], numbers[1] = numbers[1], numbers[0]
+            return numbers
+        return numbers
+    elif len(numbers) == 3:
+        return insertion_sort(numbers)
+    if len(numbers) % 2 == 0:
+        for i in range(0, len(numbers), 2):
+            temp.append(numbers[i: i + 2])
+        for pair in temp:
+            if pair[0] > pair[1]:
+                pair[0], pair[1] = pair[1], pair[0]
+    else:
+        for i in range(0, len(numbers) - 1, 2):
+            temp.append(numbers[i: i + 2])
+        for pair in temp:
+            if pair[0] > pair[1]:
+                pair[0], pair[1] = pair[1], pair[0]
+        hold_extra.append(numbers[len(numbers) - 1])
```

Comment:
`hold_extra = numbers[len(numbers) - len(numbers) % 2:]` and then you don't need that 
`if len(numbers) % 2 == 0` block.